### ISSUE TITLE
[REVIEW]: Omnizart: A General Toolbox for Automatic Music Transcription

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#3391]
**Submitting author:** @BreezeWhite (<a href="http://orcid.org/0000-0003-2362-6340">Yu Te Wu</a>)
**Repository:** <a href="https://github.com/Music-and-Culture-Technology-Lab/omnizart" target ="_blank">https://github.com/Music-and-Culture-Technology-Lab/omnizart</a>
**Version:** v0.5.0
**Editor:** @faroit
**Reviewer:** @hagenw, @keunwoochoi
**Archive:** <a href="https://doi.org/10.5281/zenodo.5769022" target="_blank">10.5281/zenodo.5769022</a>

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

## Status

[![status](https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92/status.svg)](https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92"><img src="https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92/status.svg)](https://joss.theoj.org/papers/3868f7d40bdf1e8ee77f3caa66791c92)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/Music-and-Culture-Technology-Lab/omnizart" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@hagenw & @keunwoochoi, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @faroit know.

✨ **Please start on your review when you are able, and be sure to complete your review in the next six weeks, at the very latest** ✨ 

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Music-and-Culture-Technology-Lab/omnizart">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@BreezeWhite) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] I confirm that I have read the [JOSS conflict of interest (COI) policy](https://github.com/openjournals/joss/blob/master/COI.md) and that: I have no COIs with reviewing this work or that any perceived COIs have been waived by JOSS for the purpose of this review.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/Music-and-Culture-Technology-Lab/omnizart">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@BreezeWhite) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?
- [x] **Substantial scholarly effort:** Does this submission meet the scope eligibility described in the [JOSS guidelines](https://joss.readthedocs.io/en/latest/submitting.html#substantial-scholarly-effort)

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Does the paper have a section titled 'Statement of Need' that clearly states what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @hagenw, @keunwoochoi it looks like you're currently assigned to review this paper :tada:.

**:warning: JOSS reduced service mode :warning:**

Due to the challenges of the COVID-19 pandemic, JOSS is currently operating in a "reduced service mode". You can read more about what that means in [our blog post](https://blog.joss.theoj.org/2020/05/reopening-joss).

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.32 s (586.1 files/s, 176635.4 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
YAML                            27             32             14          29650
Python                         111           2555           2981           8816
JavaScript                       3           1604           1860           6244
reStructuredText                25            432            344            587
HTML                             5             55              0            429
Markdown                         5            168              0            412
TeX                              2             30              0            264
CSS                              2             56              1            241
Bourne Shell                     4             48             30            166
TOML                             1              7              0             59
make                             2             23             17             56
DOS Batch                        1              8              1             26
JSON                             1              0              0             26
Dockerfile                       1              6              1             16
-------------------------------------------------------------------------------
SUM:                           190           5024           5249          46992
-------------------------------------------------------------------------------


Statistical information for the repository 'ddba99c9d905a23824260953' was
gathered on 2021/06/21.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
BreezeWhite                    267         19646           5818           66.32
Derek-Wu                         3             5              5            0.03
Yin-Jyun Luo                     4            79             74            0.40
unknown                         19         10147            230           27.02
yjluo                           19          1582            812            6.23

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
BreezeWhite               13450           68.5          6.3                6.22
Yin-Jyun Luo                 60           75.9          5.8               15.00
unknown                   10085           99.4          3.1               18.55
yjluo                       465           29.4          6.4                4.73
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/taslp.2020.3030482 is OK
- 10.1007/978-3-030-01234-2_49 is OK
- 10.1109/taslp.2015.2442411 is OK
- 10.1109/icassp39728.2021.9414409 is OK
- 10.1109/icassp.2018.8462079 is OK
- 10.1109/icassp.2018.8462420 is OK
- 10.1109/access.2019.2960566 is OK
- 10.1109/tpami.2018.2858821 is OK
- 10.1109/tasl.2009.2026503 is OK
- 10.1109/icassp.2018.8461686 is OK
- 10.25080/majora-7b98e3ed-003 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: :point_left:

**Comment by faroit** [comment]
👏 @hagenw @keunwoochoi - This is the review thread for the paper. All of our communications will happen here from now on. 

Both of you have checklists at the top of this thread (in that first comment) with the JOSS requirements. As you go over the submission, please check any items that you feel have been satisfied. There are also links to the JOSS reviewer guidelines.

The JOSS review is different from most other journals. Our goal is to work with the authors to help them meet our criteria instead of merely passing judgment on the submission. As such, the reviewers are encouraged to submit issues and pull requests on the software repository. When doing so, please mention https://github.com/openjournals/joss-reviews/issues/3391 so that a link is created to this thread (and I can keep an eye on what is happening). Please also feel free to comment and ask questions on this thread. In my experience, it is better to post comments/questions/suggestions as you come across them instead of waiting until you've reviewed the entire package.

**Comment by whedon** [comment]
:wave: @hagenw, please update us on how your review is going (this is an automated reminder).

**Comment by whedon** [comment]
:wave: @keunwoochoi, please update us on how your review is going (this is an automated reminder).

**Comment by hagenw** [comment]
I will submit mine tomorrow.

**Comment by faroit** [comment]
> I will submit mine tomorrow.

Great. thanks! As mentioned above, a way is to split comments/concerns into smaller issues that can more easily be approached by the authors.

**Comment by hagenw** [comment]
I cannot fill out the bullet points as the invitation under https://github.com/openjournals/joss-reviews/invitations has expired.

**Comment by danielskatz** [comment]
@whedon re-invite @hagenw as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@hagenw please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by keunwoochoi** [comment]
@danielskatz can you also invite me again? thanks!

**Comment by faroit** [comment]
@keunwoochoi Thanks for your great feedback you gave to the authors. It looks like many of your issues that you raised about to be addressed now. In the meantime can you please proceed with the checklist so that we can continue with the review process?

**Comment by keunwoochoi** [comment]
@faroit sure. can you please invite me as a reviewer so that i can edit the checkboxes?

**Comment by faroit** [comment]
@whedon re-invite @keunwoochoi as reviewer

**Comment by whedon** [comment]
OK, the reviewer has been re-invited.

@keunwoochoi please accept the invite by clicking this link: https://github.com/openjournals/joss-reviews/invitations

**Comment by faroit** [comment]
@keunwoochoi can you update us if you need more time for the review at this point – specifically proceeding with the checklist?

**Comment by keunwoochoi** [comment]
@faroit sorry that it took too long. just finished updating my checklist. i couldn't finish the review for the performance part as i do not have a direct access to any linux machine that i can easily use. the rest of the checklist items seems to be done.

**Comment by faroit** [comment]
@keunwoochoi thanks for your update. 

> i couldn't finish the review for the performance part as i do not have a direct access to any linux machine that i can easily use. the rest of the checklist items seems to be done.

Is this related to https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/38 ? 

If yes, I think it would be unfortunate to require a Linux machine to test the performance of the package. As a mac user myself, I will give it a try and try to help fixing the issue.

**Comment by keunwoochoi** [comment]
Yes it is. 

> I will give ...

That'd be great, thank you for helping my poor reviewing! 😭

**Comment by hagenw** [comment]
I'm nearly finished with the review.

The only point missing is getting the repository setup for development following the `CONTRIBUTING` guidelines (which has not worked so far for me, see https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/37), and then trying to run the tests locally and building the documentation locally.

**Comment by faroit** [comment]
@keunwoochoi your issue with macos seems to be caused by madmom. While the issues is being addressed, please check https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/38 for a workaround to run the software so that we can proceed with the review quickly.

**Comment by faroit** [comment]
@hagenw 

> The only point missing is getting the repository setup for development following the `CONTRIBUTING` guidelines (which has not worked so far for me, see [Music-and-Culture-Technology-Lab/omnizart#37](https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/37)), and then trying to run the tests locally and building the documentation locally.

seems like that this is solved. Please close the remaining checkboxes if this is the case. Thanks a lot for your review!

**Comment by hagenw** [comment]
I'm still not able to run the tests locally as the updated installation instruction does not work for me, see https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/40#issuecomment-903539742

But as I will be away for two weeks from tomorrow it might be fine if I just tick the boxes or somebody else can try to get the tests running locally.

**Comment by BreezeWhite** [comment]
Hi @faroit,

Sorry for the slow update... I have been busy working on other things and don't have much on the issues. But I will have some time recently. On the lab's side, the director is also looking for people to help on the issues, and hope I can hangover the work in near future. Still, this could take probably up to few months, I will discuss with the director and reply to you as soon as possible.

**Comment by faroit** [comment]
@BreezeWhite have you been able to find some time? Otherwise I would suggest to pause this review

**Comment by BreezeWhite** [comment]
Sorry for not having time to keep track on the review process. I think it's proper to pause the process now until the lab finds someone to keep working on the review.

**Comment by faroit** [comment]
@hagenw @keunwoochoi do you think there are remaining issues that can be set to lower priority so that they can be done later, thus having the submission published sooner and not requiring @BreezeWhite to pause the review?

**Comment by hagenw** [comment]
Would be fine with me, my main concern was that the guidelines how to run the tests locally did not work for me.
When this is ok for publication then please go ahead.

**Comment by keunwoochoi** [comment]
My issues are small documentation issues that wouldn’t be demanding so much. I’m afraid that could mean they will never be fixed if we decide it’s fine for now.
But, I wouldn’t try to convince others if we decide go ahead. All in all, this is my first time reviewing here and I’m learning, too.

**Comment by BreezeWhite** [comment]
Hi @hagenw, I've updated the code. Can you pull the code and try again?

**Comment by hagenw** [comment]
> Hi @hagenw, I've updated the code. Can you pull the code and try again?

Thanks, now everything worked as expected and I have checked all boxes in the review.

**Comment by faroit** [comment]
@keunwoochoi can you give us an update here if the remaining checkbox can be checked to proceed with the review?

**Comment by keunwoochoi** [comment]
Hi @faroit, I agree that we can proceed with this submission.

**Comment by faroit** [comment]
@whedon generate pdf

**Comment by faroit** [comment]
@whedon check repository

**Comment by faroit** [comment]
@whedon check references

**Comment by whedon** [comment]
Wordcount for `paper.md` is 1117

**Comment by whedon** [comment]
```
Software report (experimental):

github.com/AlDanial/cloc v 1.88  T=0.41 s (437.9 files/s, 71154.2 lines/s)
-------------------------------------------------------------------------------
Language                     files          blank        comment           code
-------------------------------------------------------------------------------
Python                         111           2554           2981           8782
JavaScript                       3           1604           1860           6244
YAML                            18             30             14           1770
reStructuredText                25            438            356            585
Markdown                         5            178              0            436
HTML                             5             55              0            429
TeX                              2             30              0            264
CSS                              2             57              1            244
Bourne Shell                     5             53             30            178
make                             2             25             17             68
TOML                             1              7              0             58
DOS Batch                        1              8              1             26
Dockerfile                       1              8              1             17
-------------------------------------------------------------------------------
SUM:                           181           5047           5261          19101
-------------------------------------------------------------------------------


Statistical information for the repository '721623074a973e39c5461cee' was
gathered on 2021/12/06.
The following historical commit information, by author, was found:

Author                     Commits    Insertions      Deletions    % of changes
BreezeWhite                    283         19753           5960           66.53
Derek-Wu                         3             5              5            0.03
Yin-Jyun Luo                     4            79             74            0.40
unknown                         19         10147            230           26.85
yjluo                           19          1582            812            6.19

Below are the number of rows from each author that have survived and are still
intact in the current revision:

Author                     Rows      Stability          Age       % in comments
BreezeWhite               13421           67.9         11.5                6.23
Yin-Jyun Luo                 60           75.9         11.6               15.00
unknown                   10083           99.4          8.9               18.56
yjluo                       461           29.1         12.2                4.77
```

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: :point_left:

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/taslp.2020.3030482 is OK
- 10.1007/978-3-030-01234-2_49 is OK
- 10.1109/taslp.2015.2442411 is OK
- 10.1109/icassp39728.2021.9414409 is OK
- 10.1109/icassp.2018.8462079 is OK
- 10.1109/icassp.2018.8462420 is OK
- 10.1109/access.2019.2960566 is OK
- 10.1109/tpami.2018.2858821 is OK
- 10.1109/tasl.2009.2026503 is OK
- 10.1109/icassp.2018.8461686 is OK
- 10.25080/majora-7b98e3ed-003 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by faroit** [comment]
@BreezeWhite The submission looks good now and we can finally move forward 👏 I just did check the paper again for grammar and spellings and created a pull request that you can merge in if you like.

Furthermore I found some inconsistencies in the bibliography that should be fixed as well. (see https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/49) 

When these changes are made, the paper is ready, please then please make a new release of the main repo that includes all of the changes that have resulted from the review. Please report the version number here.

Then, please make an archive of the software in Zenodo/figshare/other service and update this thread with the DOI of the archive. For the archive version, please make sure that:

- The title of the archive is the same as the JOSS paper title
- The authors of the archive are the same as the JOSS paper authors

**Comment by BreezeWhite** [comment]
Hi @faroit, thanks for the update. I've published a new version `v0.5.0` that includes all the reviewed contents. The DOI is `https://zenodo.org/record/5769022`. 
I'm not sure what you mean about the *archive title*. Do you mean the release title? And where should I refer author names to the 'archive'?

**Comment by faroit** [comment]
@BreezeWhite the archive (zenodo) looks good. Please have a look at https://github.com/Music-and-Culture-Technology-Lab/omnizart/issues/49 again before we can proceed with the final accept of this paper.

**Comment by faroit** [comment]
@whedon set v0.5.0 as version

**Comment by whedon** [comment]
OK. v0.5.0 is the version.

**Comment by faroit** [comment]
@whedon set 10.5281/zenodo.5769022 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.5769022" target="_blank">10.5281/zenodo.5769022</a> is the archive.

**Comment by faroit** [comment]
@whedon check references

**Comment by faroit** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
:point_right::page_facing_up: [Download article proof](https://raw.githubusercontent.com/openjournals/joss-papers/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: [View article proof on GitHub](https://github.com/openjournals/joss-papers/blob/joss.03391/joss.03391/10.21105.joss.03391.pdf) :page_facing_up: :point_left:

**Comment by faroit** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2803

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2803, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/taslp.2020.3030482 is OK
- 10.1007/978-3-030-01234-2_49 is OK
- 10.1109/taslp.2015.2442411 is OK
- 10.1109/icassp39728.2021.9414409 is OK
- 10.1109/icassp.2018.8462079 is OK
- 10.1109/icassp.2018.8462420 is OK
- 10.1109/access.2019.2960566 is OK
- 10.1109/TPAMI.2018.2858821 is OK
- 10.1109/tasl.2009.2026503 is OK
- 10.1109/icassp.2018.8461686 is OK
- 10.1145/2964284.2973795 is OK
- 10.25080/majora-7b98e3ed-003 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/taslp.2020.3030482 is OK
- 10.1007/978-3-030-01234-2_49 is OK
- 10.1109/taslp.2015.2442411 is OK
- 10.1109/icassp39728.2021.9414409 is OK
- 10.1109/icassp.2018.8462079 is OK
- 10.1109/icassp.2018.8462420 is OK
- 10.1109/access.2019.2960566 is OK
- 10.1109/TPAMI.2018.2858821 is OK
- 10.1109/tasl.2009.2026503 is OK
- 10.1109/icassp.2018.8461686 is OK
- 10.1145/2964284.2973795 is OK
- 10.25080/majora-7b98e3ed-003 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by danielskatz** [comment]
@BreezeWhite - I'm the AEiC who will accept and publish this, and while proofreading, I found a number of issues, addressed in https://github.com/Music-and-Culture-Technology-Lab/omnizart/pull/50 - please merge this, or let me know what you disagree with, then we can proceed.

**Comment by BreezeWhite** [comment]
@danielskatz thanks for the proofreading, I've merged the pull request.

**Comment by danielskatz** [comment]
@whedon recommend-accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
:wave: @openjournals/joss-eics, this paper is ready to be accepted and published.

 Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/2807

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/2807, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by whedon** [comment]
```
Reference check summary (note 'MISSING' DOIs are suggestions that need verification):

OK DOIs

- 10.1109/taslp.2020.3030482 is OK
- 10.1007/978-3-030-01234-2_49 is OK
- 10.1109/taslp.2015.2442411 is OK
- 10.1109/icassp39728.2021.9414409 is OK
- 10.1109/icassp.2018.8462079 is OK
- 10.1109/icassp.2018.8462420 is OK
- 10.1109/access.2019.2960566 is OK
- 10.1109/TPAMI.2018.2858821 is OK
- 10.1109/tasl.2009.2026503 is OK
- 10.1109/icassp.2018.8461686 is OK
- 10.1145/2964284.2973795 is OK
- 10.25080/majora-7b98e3ed-003 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by danielskatz** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1469122755723419649) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/2808
1. Wait a couple of minutes, then verify that the paper DOI resolves [https://doi.org/10.21105/joss.03391](https://doi.org/10.21105/joss.03391)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? Notify your editorial technical team...

**Comment by danielskatz** [comment]
Congratulations to @BreezeWhite (Yu Te Wu) and co-authors!!

And thanks to @hagenw and @keunwoochoi for reviewing, and @faroit for reviewing!
We couldn't do this without you

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.03391/status.svg)](https://doi.org/10.21105/joss.03391)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.03391">
  <img src="https://joss.theoj.org/papers/10.21105/joss.03391/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.03391/status.svg
   :target: https://doi.org/10.21105/joss.03391
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.03391/status.svg)](https://doi.org/10.21105/joss.03391)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
